### DISCUSSION TITLE
AutoGen answering only 1st of 2 mails, max consecutive auto reply value settings(?) ...

### DISCUSSION THREAD

**Discussion by stevenbaert** [microsoft/autogen#1696] [General]
Hi,

I have an AutoGen script which checks emails and answers them one by one following a script.
Works fine now, but had to change the value    max_consecutive_auto_reply=20 of the user_proxy. Else it did fetch mails but only answered first one, prepared second answer but did not send it.

So now I wonder if I would have really a lot of mails and treshold of 20 is superseded.
How can I tackle this issue? I need it to answer all mails (based upon mailinglist I allow it to answer else it gets to costly).
But need to be sure it does answer the mails (now -before changing value to 20- the 2nd mail was marked as read but answer never sent(!).

If any unclarity, do let me know.
Thanks!
Steven

---
My code till now:

`chatbot = autogen.AssistantAgent(
    name="chatbot",
    system_message="For coding tasks, only use the functions you have been provided with. Reply TERMINATE when the task is done.",
    llm_config=llm_config,
    code_execution_config={"work_dir": "scripts", "use_docker": False},
)

user_proxy = autogen.UserProxyAgent(
    name="user_proxy",
    is_termination_msg=lambda x: x.get("content", "") and x.get("content", "").rstrip().endswith("TERMINATE"),
    human_input_mode="NEVER",
    max_consecutive_auto_reply=20, #TODO AND TOCHECK !!! Heel Belangrijk, als er meerdere mails zijn, gaat hij dan stoppen na 20?
    code_execution_config={"work_dir": "scripts", "use_docker": False},
)

user_proxy.register_function(
    function_map={
        "send_email": send_mail,
        "fetch_emails": fetch_emails
    }
)

# Setup GroupChat
groupchat = autogen.GroupChat(agents=[user_proxy, chatbot], messages=[], max_round=12)

# GroupChatManager to manage the chat session
manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)`

**Comment by Lnsrini** [comment]
How about initiating a new chat for every E-mail?

**Reply by stevenbaert** [reply]
Makes absolutely sense, simple and effective. Thanks!

**Reply by stevenbaert** [reply]
Actually I don't like the idea: now the agents have the retrieve mail function. With your approach I have to use all logic outside of AutoGen. Thereby making it less flexible and not autohealing. Agents can act upon issues. 
Just struggling with the max communications, isn't there a solution for? Like setting that as a variable depending on the amount of mails?
